rocks/RocksDatabase.java

```diff
@@ -143,21 +145,20 @@ static RocksDatabase loadAndOpen(RocksTypeDB typedb, String name, Factory.Sessio
     }
 
     protected void initialise() {
-        initialiseSchema();
+        initialiseOrLoadSchema();
+        initialiseEncodingVersion();
         initialiseData();
-        mayInitRocksLogger();
         isOpen.set(true);
         try (RocksSession.Schema session = createAndOpenSession(SCHEMA, new Options.Session()).asSchema()) {
             try (RocksTransaction.Schema txn = session.initialisationTransaction()) {
                 if (txn.graph().isInitialised()) throw TypeDBException.of(DIRTY_INITIALISATION);
-                txn.schemaStorage().initialiseEncoding();
```

Comment:
we do NOT wait until the whole TypeDB is ready to valididate the encoding - we will have failed by now if we do it this late in the execution path.